forward43/kickstarter_scraper.py

```diff
@@ -0,0 +1,134 @@
+import os
+
+import json
+import random
+import time
+import requests
+from pathlib        import Path
+
+from datetime       import datetime
+from urllib.request import urlopen
+
+from fp.fp          import FreeProxy
+
+from spoof_params   import user_agent_list, accept_list
+
+
+MAX_VALUE                  = 65536
+KICKSTARTER_URL            = 'https://www.kickstarter.com/discover/advanced.json?'
+KICKSTARTER_DEFAULT_PARAMS = ['sort=popularity']
+ATTRIBUTES                 = ['rank', 'id', 'name', 'creator', 'goal', 'pledged',
+                              'state', 'backers', 'launch_date', 'deadline', 'url']
+
+def write_data_to_file(data, category):
+    '''
+    Write json data to file
+    @data: dictionary
+    @category: category id
+    '''
+    filepath = '../data/{}.json'.format(category)
+
+    with open(filepath, 'w+') as f:
+        json.dump(data, f)
+
+
+def generate_url(category, page):
+    '''
+    Generates a URL to scrape from. It's necessary that category and page parameters are not None
+    '''
+    random_seed   = 'seed='   + str(random.randint(1, MAX_VALUE))
+    random_woe_id = 'woe_id=' + str(random.randint(1, MAX_VALUE))
+    get_params  = KICKSTARTER_DEFAULT_PARAMS + ['category_id={}'.format(category), 'page={}'.format(str(page)), random_seed, random_woe_id]
+    return KICKSTARTER_URL + '&'.join(get_params)
+
+def fetch_data(url, proxies):
+
+    try:
+        user_agent_ = random.choice(user_agent_list)
+        accept_     = random.choice(accept_list)
+        headers_    = {
+            'User-Agent': user_agent_,
+            'Accept': accept_,
+            'Accept-Language': 'en-US,en;q=0.5',
+            'DNT': '1',
+            'Connection': 'keep-alive',
+            'Upgrade-Insecure-Requests': '1',
+        }
+
+        r = requests.get(url, headers=headers_, proxies=proxies)
```

Comment:
Nice one! 
completely non-need to have, but if you find yourself with nothing to do:  bonus points if you abstract this out as a wrapper for requests.get for other scrapers to use :)

Reply:
:+1:  I'm in process of making a generic scraper class with methods for data handling, processing, etc. I will include spoofing there as well.

Reply:
👍